cmd/clusterctl/client/tree/discovery_test.go

```diff
@@ -417,22 +428,29 @@ func Test_Discovery(t *testing.T) {
 				"cluster.x-k8s.io/v1beta1, Kind=Machine, ns1/cp1": {},
 				// Workers should have a machine deployment
 				"virtual.cluster.x-k8s.io/v1beta1, Kind=WorkerGroup, ns1/Workers": {
-					"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md1",
+					"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md01",
+					"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md02",
 				},
 				// Machine deployment should have a group of machines (grouping) and templates group
-				"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md1": {
+				"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md01": {
 					"virtual.cluster.x-k8s.io/v1beta1, Kind=MachineGroup, ns1/zzz_",
-					"virtual.cluster.x-k8s.io/v1beta1, Kind=TemplateGroup, ns1/md1",
+					"virtual.cluster.x-k8s.io/v1beta1, Kind=TemplateGroup, ns1/md01",
+				},
+				// MachineDeployment TemplateGroup should have a BootstrapConfigRef and InfrastructureRef
+				"virtual.cluster.x-k8s.io/v1beta1, Kind=TemplateGroup, ns1/md01": {
+					"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureMachineTemplate, ns1/md01",
+					"bootstrap.cluster.x-k8s.io/v1beta1, Kind=GenericBootstrapConfigTemplate, ns1/md01",
 				},
+				// MachineDeployment InfrastructureRef should be a leaf
+				"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureMachineTemplate, ns1/md01": {},
 				// MachineDeployment TemplateGroup should have a BootstrapConfigRef and InfrastructureRef
-				"virtual.cluster.x-k8s.io/v1beta1, Kind=TemplateGroup, ns1/md1": {
-					"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureMachineTemplate, ns1/md1",
-					"bootstrap.cluster.x-k8s.io/v1beta1, Kind=GenericBootstrapConfigTemplate, ns1/md1",
+				"virtual.cluster.x-k8s.io/v1beta1, Kind=TemplateGroup, ns1/md02": {
+					"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureMachineTemplate, ns1/md02",
 				},
 				// MachineDeployment InfrastructureRef should be a leaf
-				"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureMachineTemplate, ns1/md1": {},
+				"infrastructure.cluster.x-k8s.io/v1beta1, Kind=GenericInfrastructureMachineTemplate, ns1/md02": {},
 				// MachineDeployment BootstrapConfigRef should be a leaf
-				"bootstrap.cluster.x-k8s.io/v1beta1, Kind=GenericBootstrapConfigTemplate, ns1/md1": {},
+				"bootstrap.cluster.x-k8s.io/v1beta1, Kind=GenericBootstrapConfigTemplate, ns1/md01": {},
```

Comment:
Can you please move this below l.445 so that everything related `md01` is grouped together?